piecrust/src/imports.rs

```diff
@@ -64,7 +65,12 @@ fn c(
     let caller_remaining = instance
         .get_remaining_points()
         .expect("there should be points remaining");
-    let callee_limit = caller_remaining * POINT_PASS_PCT / 100;
+
+    let callee_limit = if points_limit > 0 && points_limit < caller_remaining {
+        points_limit
+    } else {
+        caller_remaining * POINT_PASS_PCT / 100
+    };
```

Comment:
So it is possible that the `point_limit` is larger than the 93% of the remaining points. Could that be an issue? I guess not but it might be worth to have an extra look at this.

Reply:
The idea is that technically it is up to the contract writer to choose the value appropriately. If it really wants to allow the called contract to consume most/all of its gas it should be allowed to do so.